### PR TITLE
Abaqus user docs

### PR DESCRIPTION
**Pull request by LKedward**
- Adds some basic user documentation for using Abaqus with HPC:
    - A short guide on setting up Abaqus & ifort on BC4/BP;
    - A couple of example job scripts and;
    - A troubleshooting section with common problems (based off grp-Abaqus posts).
- I added the `sphinx_design` extension to be able to use tabbed panels (not sure if there's a built-in way to do them?)

### CODE REVIEW

**Comment by LKedward** [comment]
Tagging @MattEdwards94 and @jcwomack for review

**Comment by jcwomack** [comment]
Hi @LKedward, thanks for your contribution! It looks great at a glance. I will aim to review the content soon.

On tabs in Sphinx: 

In the [ACRC HPC user documentation](https://www.acrc.bris.ac.uk/protected/hpc-docs/), the [sphinx-tabs](https://sphinx-tabs.readthedocs.io/en/latest/) extension is used. [sphinx-design](https://sphinx-design.readthedocs.io/en/furo-theme/get_started.html) seems to provide equivalent functionality, alongside some other nice features. Both are hosted by the [ExecutableBooks](https://github.com/executablebooks) project's GitHub organisation (the Executable Books project also develops the [MyST Parser](https://github.com/executablebooks/MyST-Parser).

[sphinx-design](https://sphinx-design.readthedocs.io/en/furo-theme/get_started.html) seems like a good choice for the HPC Community Docs. It will provide support for tab content as well as giving us the option of using constructs other than tabs, e.g. grids and cards.

**Comment by jcwomack** [comment]
> You can view a static preview of the compiled documentation [here](https://lkedward.github.io/HPC-community-docs/engineering/abaqus/index.html). 

Thanks for sharing a static preview. However, would it be possible to take this deployment of the docs down?

I added some information about why this is a problem to `CONTRIBUTING.md` on my fork, but have yet to merge this into the official repo:
https://github.com/jcwomack/HPC-community-docs/blob/issue/8/CONTRIBUTING.md#deployment-of-the-documentation

Essentially, having multiple published versions of the documentation on the web is likely to cause confusion to users and potentially result in duplicate entries in search engine results. To ensure there is one canonical version of the documentation, it is better to only have one repository deploying to GitHub Pages.

As an alternative, in the text linked above I describe how you can view the generated documentation for any workflow by downloading a build artifact and viewing the pages locally in a web browser.

**Comment by LKedward** [comment]
> Essentially, having multiple published versions of the documentation on the web is likely to cause confusion to users and potentially result in duplicate entries in search engine results. To ensure there is one canonical version of the documentation, it is better to only have one repository deploying to GitHub Pages.

Yes I totally understand this but, to be clear, that deployment is only meant to be temporary while this PR is reviewed.
I've found it very helpful in the past to provide deployed versions to make it easier for non-technical reviewers (and lazy technical ones!) On Fortran-Lang.org we have a workflow for maintainers to deploy prospective PRs to a subdirectory structure, _e.g._ https://fortran-lang.org/pr/347/, and I do something similar with [latex-github-collab](https://github.com/LKedward/latex-github-collab).

Now that the build artifact is generated I'll take the deployment down and point people towards the build artifact for reviewing. Cheers!

File: source/engineering/abaqus/troubleshooting.md
Comment (jcwomack) [inline source/engineering/abaqus/troubleshooting.md:51]: It is reasonable to request all physical memory on a node? I'm not sure -- do we need to leave room for the OS? I usually use `--mem=0` to request all memory available per node (and let the scheduler decide how much this is), rather than trying to specify.

Either way, I would suggest removing specific information about the memory per node and instead linking to a page in the ACRC HPC user docs with information about the hardware of the clusters (e.g. https://www.acrc.bris.ac.uk/protected/hpc-docs/training/intro-to-hpc-slurm/acrc_systems.html). Then this information does not need to be maintained in two places.

File: source/engineering/abaqus/troubleshooting.md
Comment (jcwomack) [inline source/engineering/abaqus/troubleshooting.md:70]:
```suggestion
If your jobs ends before the allotted maximum time, make sure you are running the job

```
Spelling pedantry, again 😁

File: source/engineering/abaqus/troubleshooting.md
Comment (jcwomack) [inline source/engineering/abaqus/troubleshooting.md:74]:
```suggestion
you can check with the `user-quota` command.
```

**Review by jcwomack — CHANGES_REQUESTED** [review]
Thanks for this contribution, @LKedward. I think it will be a very useful resource for Abaqus users at the university.

I have requested changes before merging, as have made a number of suggestions. All of these are minor and I am happy to discuss whether they are necessary.

File: source/engineering/abaqus/index.md
Comment (LKedward) [reply source/engineering/abaqus/index.md:43]: Cheers, this solves a problem I was having with using code blocks here!

File: source/engineering/abaqus/multi-node.md
Comment (LKedward) [reply source/engineering/abaqus/multi-node.md:45]: Yes, the `env_file` may already exist with other configuration options in it which we don't want to overwrite.

File: source/engineering/abaqus/single-node.md
Comment (LKedward) [reply source/engineering/abaqus/single-node.md:15]: Good spot! I've come across this before, but it skipped my mind here while combining two sources of information!

Since these docs are targeted at specific systems, I'll just replace it with `/usr/bin/bash -l` which we know works.
I think relying on a login shell for bash scripts is poor practice, but I believe people in the BCI may be relying on some environment customization and I don't want to induce a number of hpc help queries based on differing environments.

File: source/engineering/abaqus/troubleshooting.md
Comment (LKedward) [reply source/engineering/abaqus/troubleshooting.md:52]: Good point. I didn't know you could do `--mem=0` to request all available!
Yeah I agree, linking to the ACRC docs is a better idea. (I couldn't find this specs page originally, I was expecting this information to be at the top-level maybe under 'HPC Systems Overview'?)

File: source/engineering/abaqus/index.md
Comment (LKedward) [reply source/engineering/abaqus/index.md:24]: As far as I am aware, there are no extra steps required for HPC once a user has been granted access to the university license.

File: source/engineering/abaqus/multi-node.md
Comment (LKedward) [reply source/engineering/abaqus/multi-node.md:16]: I'm hesitant to use tabs here because the code block is big and the difference between the two systems is small (one line). Using tabs may imply that there are larger fundamental differences between the submission scripts than simply a different module name. I guess you could quickly flip between tabs to see what is different, but there is now the possibility of the tabs not being kept in sync with future changes.

Comment (LKedward) [reply source/engineering/abaqus/multi-node.md:16]: Cheers, I've now changed to `languages/intel/2020-u4`.

I couldn't find this initially with `module spider intel` because the stem has changed - I now notice it is there under `Other possible module matches`. The naming of modules on the clusters is frustratingly inconsistent.

File: source/engineering/abaqus/single-node.md
Comment (LKedward) [reply source/engineering/abaqus/single-node.md]: No, not a limitation of Abaqus. I've updated to remove this Phase 4 specific information.

File: source/engineering/abaqus/single-node.md
Comment (LKedward) [reply source/engineering/abaqus/single-node.md]: Thanks, I've now updated to reflect this

**Comment by LKedward** [comment]
Many thanks for the detailed review James @jcwomack! I've applied the suggested changes and addressed the other issues. Please see my comments for more.

File: source/engineering/abaqus/multi-node.md
Comment (jcwomack) [reply source/engineering/abaqus/multi-node.md:45]: That makes sense. However, does this mean that if you run the script twice in the same location without replacing the `env_file`, then you will get two `mp_host_list=[...]` lines? Hopefully Abaqus will use the last line in the file.

It looks like this is true according to [abaqus-docs.mit.edu](https://abaqus-docs.mit.edu/2017/English/SIMACAEILGRefMap/simailg-m-EnvUsing-sb.htm):

> If the same parameter is defined in more than one of these environment files or is defined more than once within a file, the last definition encountered will be used.

**Review by jcwomack — APPROVED** [review]
Thanks for all your work on this @LKedward! This is a great contributions and I am happy to merge it into the documentation.

File: source/engineering/abaqus/multi-node.md
Comment (LKedward) [reply source/engineering/abaqus/multi-node.md:45]: Ah I see what you were getting at. Yes the `env_file` is essentially a python script that gets sourced within Abaqus, so that isn't an issue. Ideally, researchers shouldn't be launching jobs repeatedly within the same folder due to files that Abaqus generates there. Also, the latest version of Abaci now supports launching jobs via SLURM and handles all this complication behind the scenes.

**Pull request merged**
